index.js

```diff
@@ -0,0 +1,213 @@
+var exerciseString = "This is a dummy string for testing";
+var exerciseArray = exerciseString.split(" ");
+
+console.log("Mini-assignment: week 12 day 02");
+console.log("===============================\n");
+console.log(`String to use: ${exerciseString}`);
+console.log(`Array to use: ${exerciseArray}`);
+
+console.log(`\nPOP function
+===========`);
+console.log(`Last element: ${arrayPop(exerciseArray)}`);
+console.log(`New array: ${exerciseArray}`);
+
+function arrayPop(array) {
+  let last = array[array.length - 1];
+  let temporal = [];
+  for (let i = 0; i < array.length - 1; i++) {
+    temporal.push(array[i]);
+  }
+  console.log(temporal);
+  array = temporal;
+  return last; //returns last element
+}
+
+console.log(`\nEvery function
+==============`);
+console.log(`Condition: All elements are strings`);
+console.log(
+  `Is exerciseArray all string: arrayEvery(exerciseArray, allString): ${arrayEvery(
+    exerciseArray,
+    allString
+  )}`
+);
+console.log(`Is [1,2,3,4] all string: ${arrayEvery([1, 2, 3, 4], allString)}`);
+
+function arrayEvery(array, anotherFunction) {
+  for (let i = 0; i < array.length - 1; i++) {
+    if (!anotherFunction(array[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+function allString(elem) {
+  return typeof elem == "string";
+}
+console.log(`\nSome function
+=============`);
+console.log(`Condition: All elements are strings`);
+console.log(
+  `Has exerciseArray any numbers: arraySome(exerciseArray, allNumber): ${arraySome(
+    exerciseArray,
+    allNumber
+  )}`
+);
+console.log(`Has [1,2,3,4] any numbers: ${arraySome([1, 2, 3, 4], allNumber)}`);
+console.log(
+  `Has [1,"2",3,4] any number: ${arraySome([1, 2, 3, 4], allNumber)}`
+);
+
+function arraySome(array, anotherFunction) {
+  for (let i = 0; i < array.length - 1; i++) {
+    if (anotherFunction(array[i])) {
+      return true;
+    }
+  }
+  return false;
+}
+function allNumber(elem) {
+  return typeof elem == "number";
+}
+console.log(`\nFind function
+=============`);
+console.log(`Condition: Find first number > 10`);
+console.log(
+  `Show first number that fulfills condition from exerciseArray: arrayFind(exerciseArray, firstNumber): ${arrayFind(
+    exerciseArray,
+    firstNumber
+  )}`
+);
+console.log(
+  `Show first number from ["1","2",3,40]: ${arrayFind(
+    ["1", "2", 3, 40],
+    firstNumber
+  )}`
+);
+
+function arrayFind(array, anotherFunction) {
+  for (let i=0; i < array.length; i++) {
+    if (anotherFunction(array[i])) {
+      return array[i];
+    }
+  }
+  void 0;
```

Comment:
can you explain me how 'void' works please? because if It is deleted the function stills working.

Reply:
it just returns undefined which is the requirement of the exercise.